src/Service/SynchronizationService.php

```diff
@@ -88,24 +104,25 @@ public function setStyle(SymfonyStyle $style): self
     }//end setStyle()
 
     /**
-     * Temporary function as replacement of the $this->oldSyncService->synchronize() function.
-     * Because currently synchronize function can only pull from a source and not push to a source.
+     * This function was created because currently $commonGateway->synchronizationService->synchronize() function
+     * can only pull from a source and not push to a source.
+     * This function can be used as 'temporary' replacement of the synchronize function.
      *
-     * @todo: Temp way of doing this without updating the oldSyncService->synchronize() function...
+     * NOTE: Before calling this function a Synchronization object must exist or be created, please use the
+     * $commonGateway->synchronizationService->findSyncBySource() or findSyncByObject() function for this.
      *
-     * @param Synchronization|null $synchronization The synchronization we are going to synchronize.
-     * @param array                $objectArray     The object data we are going to synchronize.
-     * @param ObjectEntity         $objectEntity    The objectEntity which data we are going to synchronize.
-     * @param Schema               $schema          The schema the object we are going to send belongs to.
-     * @param string               $location        The path/endpoint we send the request to.
-     * @param string|null          $idLocation      The location of the id in the response body.
-     * @param string|null          $method          The request method PUT or POST.
+     * @param Synchronization $synchronization The synchronization we are going to synchronize.
+     * @param array           $objectArray     The object data we are going to synchronize.
+     * @param ObjectEntity    $objectEntity    The objectEntity which data we are going to synchronize.
+     * @param string          $location        The path/endpoint we send the request to.
+     * @param string|null     $idLocation      The location of the id in the response body.
+     * @param string|null     $method          The request method PUT or POST.
      *
      * @return array The response body of the outgoing call, or an empty array on error.
      *
      * @throws Exception
      */
-    public function synchronizeTemp(?Synchronization &$synchronization = null, array $objectArray, ObjectEntity $objectEntity, Schema $schema, string $location, ?string $idLocation = null, ?string $method = 'POST'): array
+    public function synchronizeTemp(Synchronization $synchronization, array $objectArray, ObjectEntity $objectEntity, string $location, ?string $idLocation = null, ?string $method = 'POST'): array
```

Comment:
Ik zou die & voor $synchronization wel houden want return $body returend vaak null omdat door guzzle je vaak maar 1x de body kan gebruiken voordat deze om een gekke redden null wordt.  IK gebruik de $synchronization ook om te checken of de sync goed gegaan is en anders moet ik de sync opnieuw fetchen in mijn code bijvoorbeeld